addons/account/wizard/account_automatic_entry_wizard.py

```diff
@@ -0,0 +1,388 @@
+
+        # compute the account.move.lines and the total amount per move
+        for aml in self.move_line_ids:
+            # account.move.line data
+            reported_debit = aml.company_id.currency_id.round((self.percentage / 100) * aml.debit)
+            reported_credit = aml.company_id.currency_id.round((self.percentage / 100) * aml.credit)
+            if aml.currency_id:
+                reported_amount_currency = aml.currency_id.round((self.percentage / 100) * aml.amount_currency)
+            else:
+                reported_amount_currency = 0.0
+
+            move_data['new_date']['line_ids'] += [
+                (0, 0, {
+                    'name': aml.name,
+                    'debit': reported_debit,
+                    'credit': reported_credit,
+                    'amount_currency': reported_amount_currency,
+                    'currency_id': aml.currency_id.id,
+                    'account_id': aml.account_id.id,
+                    'partner_id': aml.partner_id.id,
+                }),
+                (0, 0, {
+                    'name': _('Adjusting Entry'),
+                    'debit': reported_credit,
+                    'credit': reported_debit,
+                    'amount_currency': -reported_amount_currency,
+                    'currency_id': aml.currency_id.id,
+                    'account_id': accrual_account.id,
+                    'partner_id': aml.partner_id.id,
+                }),
+            ]
+            move_data[aml.move_id.date]['line_ids'] += [
+                (0, 0, {
+                    'name': aml.name,
+                    'debit': reported_credit,
+                    'credit': reported_debit,
+                    'amount_currency': -reported_amount_currency,
+                    'currency_id': aml.currency_id.id,
+                    'account_id': aml.account_id.id,
+                    'partner_id': aml.partner_id.id,
+                }),
+                (0, 0, {
+                    'name': _('Adjusting Entry'),
+                    'debit': reported_debit,
+                    'credit': reported_credit,
+                    'amount_currency': reported_amount_currency,
+                    'currency_id': aml.currency_id.id,
+                    'account_id': accrual_account.id,
+                    'partner_id': aml.partner_id.id,
+                }),
+            ]
+
+        move_vals = [m for m in move_data.values()]
+        return move_vals
+
+    @api.depends('move_line_ids', 'journal_id', 'journal_id', 'revenue_accrual_account', 'expense_accrual_account', 'percentage', 'date', 'account_type', 'action', 'destination_account_id')
+    def _compute_move_data(self):
+        for record in self:
+            if record.action == 'change_period':
+                if any(line.account_id.user_type_id != record.move_line_ids[0].account_id.user_type_id for line in record.move_line_ids):
+                    raise UserError(_('All accounts on the lines must be of the same type.'))
```

Comment:
you should not raise a `UserError` inside a compute. Add a constraint instead

Reply:
No, because the `@api.constrains` is checked only when validating the record. I want the warning as soon as I change the `action` selection field.

Reply:
Check this in `default_get` then?

Reply:
> Check this in `default_get`

I can't because it depends on the action and it still can change later

> Or inside an `onchange`

I think it is even worse because it can change the action, then raise but keeping the action that we don't want

> Or prevent switching the action if the accounts are different ?

Well, it's what I am trying to do, but I don't see any better way to do it.

Why do you think it is a bad idea to do it in the compute function? It will block a `create` or a `write` with wrong values too, and it should... (`onchange` and `default_get` won't)

Reply:
Checked with RCO, it's fine